code/_onclick/hud/parallax.dm

```diff
@@ -0,0 +1,254 @@
+/client
+	var/list/parallax_layers = list()
+	var/obj/screen/parallax_pmaster/parallax_pmaster
+	var/obj/screen/parallax_space_whitifier/parallax_space_whitifier
+	var/obj/screen/parallax_fixer/parallax_fixer
+	var/atom/movable/movingmob
+	var/turf/previous_turf
+	var/do_smoothing = 1
+	var/looping_mode = 0
+	var/last_parallax_shift
+	var/parallax_movedir = 0
+
+/datum/hud/proc/create_parallax()
+	var/client/C = mymob.client
+	if(!C.parallax_layers.len)
+		C.parallax_layers += new /obj/screen/parallax_layer/layer_1
+		C.parallax_layers += new /obj/screen/parallax_layer/layer_2
+		C.parallax_pmaster = new
+		C.parallax_space_whitifier = new
+		C.parallax_fixer = new
+	for(var/obj/O in C.parallax_layers)
+		C.screen |= O
+		C.screen |= C.parallax_pmaster
+		C.screen |= C.parallax_space_whitifier
+		C.screen |= C.parallax_fixer
+
+// This sets which way the current shuttle is moving
+/datum/hud/proc/set_parallax_movedir(new_parallax_movedir)
+	var/client/C = mymob.client
+	if(new_parallax_movedir == C.parallax_movedir)
+		return
+	if(new_parallax_movedir == 0)
+		for(var/obj/screen/parallax_layer/L in C.parallax_layers)
+			animate(L)
+			L.transform = matrix()
+			L.icon_state = initial(L.icon_state)
+			L.update_o()
+		C.do_smoothing = 1
+	else
+		if(new_parallax_movedir == 4 || new_parallax_movedir == 1)
+			C.looping_mode = 1
+		else
+			C.looping_mode = 2
```

Comment:
I FUCKING FIGURED THIS OUT!

The animation only has two directions, so this flips the animation if it's the reverse of one of the them